lib/pure/asynchttpserver.nim

```diff
@@ -184,7 +184,10 @@ proc processRequest(
   # \n
   request.headers.clear()
   request.body = ""
-  request.hostname.shallowCopy(address)
+  when defined(gcArc) or defined(gcOrc):
+    request.hostname = address
```

Comment:
Make `address` a sink parameter so that this can be turned into a move by the compiler.

Reply:
thanks!